shell/src/main/java/alluxio/cli/LogLevel.java

```diff
@@ -0,0 +1,242 @@
+    CommandLineParser parser = new DefaultParser();
+    CommandLine cmd;
+
+    try {
+      cmd = parser.parse(OPTIONS, args, true /* stopAtNonOption */);
+    } catch (ParseException e) {
+      printHelp("Unable to parse input args: " + e.getMessage());
+      return 1;
+    }
+    List<TargetInfo> targetInfoList = new ArrayList<>();
+
+    String[] targets = parseOptTarget(cmd);
+    String logName = parseOptLogName(cmd);
+    String level = parseOptLevel(cmd);
+
+    for (String target : targets) {
+      if (target.contains("master")) {
+        String masterHost = NetworkAddressUtils.getConnectHost(ServiceType.MASTER_WEB);
+        int masterPort = NetworkAddressUtils.getPort(ServiceType.MASTER_WEB);
+        targetInfoList.add(new TargetInfo(masterHost, masterPort, "master"));
+      } else if (target.contains("workers")) {
+        AlluxioBlockStore alluxioBlockStore = AlluxioBlockStore.create();
+        try {
+          List<BlockWorkerInfo> workerInfoList = alluxioBlockStore.getWorkerInfoList();
+          for (BlockWorkerInfo workerInfo : workerInfoList) {
+            WorkerNetAddress netAddress = workerInfo.getNetAddress();
+            targetInfoList.add(
+                new TargetInfo(netAddress.getHost(), netAddress.getWebPort(), "worker"));
+          }
+        } catch (IOException e) {
+          e.printStackTrace();
+          return 1;
+        }
+      } else if (target.contains(":")) {
+        String[] hostPortPair = target.split(":");
+        int port = Integer.parseInt(hostPortPair[1]);
+        targetInfoList.add(new TargetInfo(hostPortPair[0], port, "worker"));
+      }
+    }
+
+    for (TargetInfo targetInfo : targetInfoList) {
+      if (handleTarget(logName, level, targetInfo)) {
+        return 1;
+      }
+    }
+    return 0;
+  }
+
+  private static String[] parseOptTarget(CommandLine cmd) {
+    String[] targets;
+    if (cmd.hasOption(TARGET_OPTION_NAME)) {
+      String argTarget = cmd.getOptionValue(TARGET_OPTION_NAME);
+      if (StringUtils.isBlank(argTarget)) {
+        targets = new String[]{"master", "workers"};
+      } else if (argTarget.contains(TARGET_SEPARATOR)) {
+        targets = argTarget.split(TARGET_SEPARATOR);
+      } else {
+        targets = new String[]{argTarget};
+      }
+    } else {
+      targets = new String[]{"master", "workers"};
+    }
+    return targets;
+  }
+
+  private static String parseOptLogName(CommandLine cmd) {
+    String argName = cmd.getOptionValue(LOG_NAME_OPTION_NAME);
+    if (StringUtils.isNotBlank(argName)) {
+      return argName;
+    }
+    return "";
+  }
+
+  private static String parseOptLevel(CommandLine cmd) {
+    if (cmd.hasOption(LEVEL_OPTION_NAME)) {
+      String argLevel = cmd.getOptionValue(LEVEL_OPTION_NAME);
+      if (StringUtils.isNotBlank(argLevel)) {
+        return argLevel;
+      }
+    }
+    return null;
+  }
+
+  private static boolean handleTarget(String logName, String level, TargetInfo targetInfo) {
```

Comment:
logically, the `target` should be the first parameter, then the logger name on that target, then the level of the log